mesonbuild/mesonlib.py

```diff
@@ -982,49 +1036,57 @@ def replace_if_different(dst, dst_tmp):
     else:
         os.unlink(dst_tmp)
 
-def listify(item: T.Any,
-            flatten: bool = True,
-            unholder: bool = False) -> T.List[T.Any]:
+
+@T.overload
+def unholder(item: 'ObjectHolder[_T]') -> _T: ...
+
+@T.overload
+def unholder(item: T.List['ObjectHolder[_T]']) -> T.List[_T]: ...
```

Comment:
You might consider using `Iterable` for the input. Then is more flexible and provides a guarantee to the caller that the list won't be modified. If `Iterable` doesn't work then `Sequence` usually does.